pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,450 @@
+                                      })
+                .always(() => {
+                    this.updateStatusFromBus();
+                })
+                .done(() => {
+                    this.setState({ problemState: ProblemState.REPORTED, });
+                })
+                .fail((ex) => {
+                    this.setState({ problemState: ProblemState.REPORTABLE, });
+                })
+                .stream((data) => {
+                    this.setState({ message: data, });
+                });
+
+        this.setState({
+            problemState: ProblemState.REPORTING,
+            process: process,
+        });
+    }
+
+    render() {
+        return <WorkflowRow label={_("Report to ABRT Analytics")}
+                            message={this.state.message}
+                            onCancelButtonClick={this.onCancelButtonClick}
+                            onReportButtonClick={this.onReportButtonClick}
+                            problemState={this.state.problemState}
+                            reportLinks={this.state.reportLinks} />;
+    }
+}
+
+const reportd_client = cockpit.defer();
+const permission = cockpit.permission({ admin: true });
+permission.addEventListener("changed", () => {
+    // reportd may use ABRT API that requires authorization, but it cannot
+    // be given using polkit, as the calling process will always be reportd
+    // and not cockpit-bridge. However, UID 0 is always authorized, hence
+    // the spawning of the system service and using the system bus here.
+    //
+    // TODO: Only use system bus when reportd is merged into ABRT
+    //       (https://github.com/abrt/reportd/issues/8).
+    const bus_type = permission.allowed ? "system" : "session";
+    const client = cockpit.dbus("org.freedesktop.reportd",
+                                {
+                                    bus: bus_type,
+                                    track: true,
+                                });
+    client.wait()
+            .done(() => {
+                reportd_client.resolve(client);
+            })
+            .fail(() => {
+                reportd_client.reject();
+            });
+});
+
+class BusWorkflowRow extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            label: this.props.workflow[1],
+            message: "",
+            problemState: ProblemState.REPORTABLE,
+            reportLinks: [],
+            task: null,
+        };
+
+        this.onCancelButtonClick = this.onCancelButtonClick.bind(this);
+        this.onReportButtonClick = this.onReportButtonClick.bind(this);
+        this.updateStatusFromBus = this.updateStatusFromBus.bind(this);
+
+        this.updateStatusFromBus();
+    }
+
+    onCancelButtonClick(event) {
+        this.state.task.Cancel();
+    }
+
+    onReportButtonClick(event) {
+        if (this.state.problemState === ProblemState.REPORTING) {
+            return;
+        }
```

Comment:
Same issue as above -- please disable buttons while there is an ongoing action, rather than ignoring the click.